src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -14,7 +14,13 @@ public MyHashMap() {
     @Override
     public void put(K key, V value) {
         if (key == null) {
-            putForNullKey(value);
+            for (Node<K, V> entry = table[0]; entry != null; entry = entry.next) {
+                if (entry.key == null) {
+                    entry.value = value;
+                    return;
+                }
+            }
+            addEntry(0, null, value, 0);
```

Comment:
Here and in get method we do not need special code for null keys. The only thing we need to do is to adjust this line of code

`if (key.equals(entry.key))`